File: AspNet/StoreApi/StoreApi/Models/OrderModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace StoreApi.Models
{
    public class OrderModel
    {
        public int Id { get; set; }

        public string Comment { get; set; }

        public int StatusId { get; set; }

        public string Status { get; set; }

        public int CustomerId { get; set; }

        

        public CustomerModel Customer { get; set; }

        public DateTimeOffset CreatedDate { get; set; }
    }
}

File: src/app/shared/models/order.model.ts
import { Customer } from './customer.model';
import { Product } from './product.model';

export class Order {
	constructor(
		public id: number,
		public createdDate: Date,
		public customer: Customer | null,
		public status: string,
		public statusId: number,
		public totalCost: number = 0,
		public products: Product[] = [],
		public comment: string = '',
		public customerId: number = 0
	) {}
}
